src/server/journal/journal_slice.h

```diff
@@ -44,6 +44,9 @@ class JournalSlice {
 
   uint32_t RegisterOnChange(ChangeCallback cb);
   void UnregisterOnChange(uint32_t);
+  bool HasRegisteredCallbacks() const {
+    return !change_cb_arr_.empty();
```

Comment:
dont we need the lock here?

Reply:
Might I suggest `ABSL_GUARDED_BY` for these cases exactly?
Abseil provides thread annotations, enforced by (at least) clang. You can read more about them here: https://abseil.io/docs/cpp/guides/synchronization#thread-annotations
I just used one in my latest PR.

Reply:
Done. I'm not super happy about adding a lock there, but I did profiling and it was something like 0.1% overhead so I'm not going to optimize because of that.